test/common/grpc/common_test.cc

```diff
@@ -349,5 +349,15 @@ TEST(GrpcCommonTest, ValidateResponse) {
   }
 }
 
+TEST(GrpcCommonTest, MakeBufferInstance) {
```

Comment:
more tests for this? e.g. buffer with a few slices, validating they are same. (same for below)

Reply:
When we are good with the API I'll break out the common changes into a separate PR and add more tests.

Reply:
done